source/grid/grid_in.cc

```diff
@@ -1451,14 +1452,73 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
           in >> line;
         }
 
-      // if the next block is of kind $Entities, ignore it
+      // if the next block is of kind $Entities, parse it
       if (line == "$Entities")
         {
-          do
+          unsigned long numPoints, numCurves, numSurfaces, numVolumes,
+            numPhysicals;
+          int    tag, physicalTag;
+          double boxMinX, boxMinY, boxMinZ, boxMaxX, boxMaxY, boxMaxZ;
+
+          in >> numPoints >> numCurves >> numSurfaces >> numVolumes;
+          for (unsigned int i = 0; i < numPoints; ++i)
             {
-              in >> line;
+              // parse point ids
+              in >> tag >> boxMinX >> boxMinY >> boxMinZ >> boxMaxX >>
+                boxMaxY >> boxMaxZ >> numPhysicals;
+              AssertThrow(numPhysicals < 2,
+                          ExcMessage("More than one tag is not supported!"));
+              for (unsigned int j = 0; j < numPhysicals; ++j)
+                in >> physicalTag;
+              tag_maps[0][tag] = (numPhysicals == 0) ? 0 : physicalTag;
+            }
+          for (unsigned int i = 0; i < numCurves; ++i)
+            {
+              // parse curve ids
+              in >> tag >> boxMinX >> boxMinY >> boxMinZ >> boxMaxX >>
+                boxMaxY >> boxMaxZ >> numPhysicals;
+              AssertThrow(numPhysicals < 2,
+                          ExcMessage("More than one tag is not supported!"));
+              for (unsigned int j = 0; j < numPhysicals; ++j)
+                in >> physicalTag;
+              tag_maps[1][tag] = (numPhysicals == 0) ? 0 : physicalTag;
+              in >> numPoints;
+              for (unsigned int j = 0; j < numPoints; ++j)
+                in >> tag;
```

Comment:
Why do we ignore these tags?

Reply:
I don't know what to do with them. The one we care for is `physical_tag`.